File: java/src/main/java/fr/arolla/Orchestrator.java
package fr.arolla;

public class Orchestrator {
  private final Booking booking;

  private final Inventory inventory;
  private final Ticketing ticketing;
  private final Notifier notifier;

  public Orchestrator(Booking booking, Inventory inventory, Ticketing ticketing, Notifier notifier) {
    this.ticketing = ticketing;
    this.inventory = inventory;
    this.booking = booking;
    this.notifier = notifier;
  }

  public void orchestrate(BookingRequest request) {
    int numSeats = request.numSeats();
    String user = request.user();

    booking.book(numSeats);
    inventory.decrementCapacity(numSeats);
    ticketing.printTicket(numSeats);
    notifier.notifyUser(user, "Your tickets are ready");
  }
}

File: dotnet/Choreography/Orchestrator.cs
namespace Choreography
{


    public class Orchestrator
    {
        private readonly Booking booking;
        private readonly Inventory inventory;
        private readonly Ticketing ticketing;

        public Orchestrator(Booking booking, Inventory inventory, Ticketing ticketing)
        {
            this.booking = booking;
            this.inventory = inventory;
            this.ticketing = ticketing;
        }

        public void Book(int numSeats)
        {
            booking.Book(numSeats);
            inventory.DecrementCapacity(numSeats);
            ticketing.PrintTickets(numSeats);
        }
    }
}
